src/Core/Domain/Profile/Permission/QueryResult/ConfigurablePermissions.php

```diff
@@ -0,0 +1,226 @@
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Profile\Permission\QueryResult;
+
+use PrestaShop\PrestaShop\Core\Domain\Profile\Permission\ValueObject\Permission;
+
+class ConfigurablePermissions
+{
+    /**
+     * @var array
+     */
+    private $profilePermissionsForTabs;
+
+    /**
+     * @var array
+     */
+    private $profiles;
+
+    /**
+     * @var array
+     */
+    private $tabs;
+
+    /**
+     * @var array
+     */
+    private $bulkConfiguration;
+
+    /**
+     * @var array
+     */
+    private $profilePermissionsForModules;
+
+    /**
+     * @var array
+     */
+    private $permissions;
+
+    /**
+     * @var array
+     */
+    private $permissionIds;
+
+    /**
+     * @var int
+     */
+    private $employeeProfileId;
+
+    /**
+     * @var bool
+     */
+    private $hasEmployeeEditPermission;
+
+    /**
+     * @param array $profilePermissionsForTabs
+     * @param array $profilePermissionsForModules
+     * @param array $profiles
+     * @param array $tabs
+     * @param array $bulkConfiguration
+     * @param string[] $permissions
+     * @param int[] $permissionIds
+     * @param int $employeeProfileId
+     * @param bool $hasEmployeeEditPermission
+     */
+    public function __construct(
+        array $profilePermissionsForTabs,
+        array $profilePermissionsForModules,
+        array $profiles,
+        array $tabs,
+        array $bulkConfiguration,
+        array $permissions,
+        array $permissionIds,
+        int $employeeProfileId,
+        bool $hasEmployeeEditPermission
+    ) {
+        $this->profilePermissionsForTabs = $profilePermissionsForTabs;
+        $this->profiles = $profiles;
+        $this->tabs = $tabs;
+        $this->bulkConfiguration = $bulkConfiguration;
+        $this->profilePermissionsForModules = $profilePermissionsForModules;
+        $this->permissions = $permissions;
+        $this->permissionIds = $permissionIds;
+        $this->employeeProfileId = $employeeProfileId;
+        $this->hasEmployeeEditPermission = $hasEmployeeEditPermission;
+    }
+
+    /**
+     * @return array
+     */
+    public function getProfilePermissionsForTabs(): array
+    {
+        return $this->profilePermissionsForTabs;
+    }
+
+    /**
+     * @return array
+     */
+    public function getProfiles(): array
+    {
+        return $this->profiles;
+    }
+
+    /**
+     * @return array
+     */
+    public function getTabs(): array
+    {
+        return $this->tabs;
+    }
+
+    /**
+     * @param int $profileId
+     *
+     * @return bool
+     */
+    public function isBulkViewConfigurationEnabled(int $profileId): bool
+    {
+        return $this->bulkConfiguration[$profileId][Permission::VIEW];
```

Comment:
Need an isset ?

Reply:
idem (bottom methods)

Reply:
Fields are mandatory and tested in `GetPermissionForConfigurationHandler`

Reply:
:+1: